### PR TITLE
[Docs] Add manifest CMake integration and eval order details

### PR DESCRIPTION
**Pull request by ihnorton**
This PR adds a paragraph to the manifest docs, detailing the CMake integration and evaluation sequence.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Personally I think that this is missing a hint about the fact that there are other ways to include vcpkg into cmake. The important part is however that it needs to be integrated before the very first `find_x` call.
Furthermore it should be noted that changing the triplet requires a clean reconfigure since the results of `find_x` calls are stored in cache variables which won't get refreshed.

**Comment by ihnorton** [comment]
Hi @Neumann-A, thanks for the feedback.

> I think that this is missing a hint about the fact that there are other ways to include vcpkg into cmake.

I added a link to the [Using with cmake](https://github.com/microsoft/vcpkg#using-vcpkg-with-cmake) instructions in the main README. In the context of manifest mode, is there another approach than setting the toolchain file -- if so, could you give me a hint/pointer what you mean?

> Furthermore it should be noted that changing the triplet requires a clean reconfigure since the results of find_x calls are stored in cache variables which won't get refreshed.

Added, thanks. (as an side, one thing I am quite curious about is what other conditions require reconfigure or trigger rebuilds -- trying to understand what a local development workflow across multiple packages would look like, for example)

**Review by Cheney-W — APPROVED** [review]

File: docs/users/manifests.md [hunk 1]
```diff
@@ -40,6 +41,10 @@ manifest mode.
   - [`"default-features"`](#default-features)
   - [`"vcpkg-configuration"`](#vcpkg-configuration)
 
+## How does it work?
+
+In manifest mode, vcpkg is integrated with CMake via a [toolchain](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html), which must be [specified](https://github.com/microsoft/vcpkg#using-vcpkg-with-cmake) when the project is configured. The toolchain file is evaluated by CMake via [`include()`](https://cmake.org/cmake/help/latest/command/include.html) the first time CMake's [`project()`](https://cmake.org/cmake/help/latest/command/project.html) command is called. This evaluation sequence means that any vcpkg CMake variables must be set before the first call to `project()`. Note that changing the triplet requires a clean reconfigure since the results of `find_*` calls are stored in cache variables which won't get refreshed. During configuration, the toolchain CMake file effectively runs `vcpkg install`, with a project-local build tree, thus executing full builds of all manifest dependencies _during CMake configuration_.
```
Comment (dg0yt) [inline docs/users/manifests.md:46]: If this was broken into multiple lines, it might be easier to comment in GH.

**Review by dg0yt — COMMENTED** [review]
The new paragraph comes with a general headline and describes CMake integration. Does this sufficiently reflect "How does it work?" AFAIU there is at least Visual Studio integration, and there might be others in the future.

I'm also wondering how this can be extended to cover the key difference between command line manifest mode and integrated manifest mode: Where is the installed tree?

**Comment by BillyONeal** [comment]
I'm dropping reviewed from these because I'm not sure how they are affected by the ongoing transition to learn.microsoft.com. /cc @ras0219-msft @AugP

**Comment by Cheney-W** [comment]
Close this PR and please track it with https://github.com/microsoft/vcpkg-docs/pull/33.

**Comment by Cheney-W** [comment]
If you have any new documentation-related issues or prs, please submit them on https://github.com/microsoft/vcpkg-docs.

**Pull request closed without merging**